### PR TITLE
packages upgraded

### PR DESCRIPTION
**Pull request by hasanm08**
## :pencil: Checklist

- [x] I reviewed submitted code
- [x] I added tests to verify changes
- [x] No new PII added or SDK only sends newly added PII if `sendDefaultPii` is enabled
- [x] I updated the docs if needed
- [x] All tests passing
- [x] No breaking changes

### CODE REVIEW

**Comment by mishkov** [comment]
Important PR to me because I cannot use latest version of other packages due to `sentry_flutter`. So for now I am just gonna depend on this PR

**Comment by kuhnroyal** [comment]
Sadly this is not how package updates work for well adapted and used packages.

**Comment by hasanm08** [comment]
> Sadly this is not how package updates work for well adapted and used packages.

how should I upgrade them when no breaking change happened?

**Comment by kuhnroyal** [comment]
You can locally always run `pub upgrade`.
For major releases you can override them if there is no breaking change until the sentry package supports the new version. 
But sentry supports several versions back and older Dart SDKs so an update should always try to increase the version range and not force the new major version on everyone.
See https://github.com/getsentry/sentry-dart/pull/1948

**Comment by hasanm08** [comment]
> You can locally always run `pub upgrade`. For major releases you can override them if there is no breaking change until the sentry package supports the new version. But sentry supports several versions back and older Dart SDKs so an update should always try to increase the version range and not force the new major version on everyone. See #1948

but it forced package_info_plus version before and I can not use it in my project because I needed package_info_plus latest version to solve the gradle 8.0 issues that is needed in flutter 3.19

**Comment by kuhnroyal** [comment]
#1948 was merged, you can check how it was solved.
You can locally use something like this:

```yaml
dependency_overrides:
  package_info_plus: 7.0.0
  ````

**Comment by hasanm08** [comment]
> #1948 was merged, you can check how it was solved. You can locally use something like this:
> 
> ```yaml
> dependency_overrides:
>   package_info_plus: 7.0.0
> ```

issue fixed recheck please

**Comment by hasanm08** [comment]
@mishkov @kuhnroyal  @benvinegar @mitsuhiko please merge it i need it

**Comment by kahest** [comment]
@hasanm08 we merged the required change for `package_info_plus` in https://github.com/getsentry/sentry-dart/pull/1948, the release for this is incoming. Thanks for opening this PR and apologies that this went under our radar - in future, please feel free to ping me to get attention on your issues/PRs.
@buenaflor please take a look if the other changes make sense

**Review by buenaflor — COMMENTED** [review]
generally looks okay to me but as far as I can see most of the changes here are applied to examples so they don't really have an impact on the SDK itself. 

have you encountered any issues here when running the samples or any specific reason you chose these packages to change the dep ranges? @hasanm08

**Comment by hasanm08** [comment]
> @hasanm08 we merged the required change for `package_info_plus` in https://github.com/getsentry/sentry-dart/pull/1948, the release for this is incoming. Thanks for opening this PR and apologies that this went under our radar - in future, please feel free to ping me to get attention on your issues/PRs.
> @buenaflor please take a look if the other changes make sense

My forked branch is uptodate with your branch but still some packages had issue
 And still packageInfoPlus needed to be upgraded in sentry_flutter
And this issue distracts my work
Please recheck and merge it

**Comment by hasanm08** [comment]
> generally looks okay to me but as far as I can see most of the changes here are either dev dependencies, or applied to examples so they don't really have an impact on the SDK itself. 
> 
> have you encountered any issues here when running the samples or any specific reason you chose these packages to change the dep ranges? @hasanm08 

I have issues using sentry_flutter (latest)and wakelock_plus (latest)
Together because of the packages version in the latest package of yours
Nevermind if you don't merge my work, but please fix this issue and publish a new version

**Comment by buenaflor** [comment]
@hasanm08 the only change that you did here that might affect you directly must be the hive pubspec.yaml which adds a lint range is that correct? 

I'm not against merging this but I wanna get the context behind it first 

> I have issues using sentry_flutter (latest)and wakelock_plus (latest)
> Together because of the packages version in the latest package of yours

We already know that package_info_plus is causing issues and are releasing a fix for it soon. 
Ideally you can provide other errors here you are encountering so we know which dependencies exactly are the problem.

**Comment by buenaflor** [comment]
just an fyi, the fix for package info is out as of `7.19.0` @hasanm08

**Comment by buenaflor** [comment]
@hasanm08 do you still need these changes?

**Comment by hasanm08** [comment]
> @hasanm08 do you still need these changes? 

Yes please merge it to use latest version

**Comment by hasanm08** [comment]
> just an fyi, the fix for package info is out as of `7.19.0` @hasanm08 

Fixed

**Comment by hasanm08** [comment]
@mishkov

**Comment by buenaflor** [comment]
> Ideally you can provide other errors here you are encountering so we know which dependencies exactly are the problem.

Can you provide errors that happen without this PR?

**Comment by hasanm08** [comment]
> > Ideally you can provide other errors here you are encountering so we know which dependencies exactly are the problem.
> 
> Can you provide errors that happen without this PR?

package info plus versioning latest is 8.0.0
please merge it to fix

File: min_version_test/pubspec.yaml [hunk 1]
```diff
@@ -18,7 +18,7 @@ publish_to: 'none' # Remove this line if you wish to publish to pub.dev
 version: 1.0.0+1
 
 environment:
-  sdk: '2.17.0'
+  sdk: '>=2.17.1 <4.0.0'
```
Comment (vaind) [inline min_version_test/pubspec.yaml:21]: The `min_version_test` project is specifically created to test against the lowest supported SDK version (2.17.0), so this change doesn't seem right. Why is it necessary?

Comment (hasanm08) [reply min_version_test/pubspec.yaml:21]: Fixed

**Comment by buenaflor** [comment]
> package info plus versioning latest is 8.0.0

this is already fixed, I don't think we need this pr anymore

**Comment by buenaflor** [comment]
Closing this pr as the problems mentioned have already been fixed.

Feel free to ping me if there is anything else

**Pull request closed without merging**